### PR TITLE
validate password during email update

### LINKED ISSUE #812: Password validation should be enforced while changing an email
**Environment**
Preview/Chrome

**To reproduce**
- Enter an incorrect password so as to change your email 
- The email changes regardless of entering an incorrect password

**Expected result**
- Password should be correct before an email is changed.

### PR DESCRIPTION
**Pull request by denniswambua**
closes #812

### CODE REVIEW

File: onadata/libs/serializers/user_profile_serializer.py [hunk 1]
```diff
@@ -147,11 +147,14 @@ def _get_params(self, attrs):
     def update(self, instance, validated_data):
         params = validated_data
 
-        # Check password if email is beig updated
+        # Check password if email is being updated
         if 'email' in params and 'password1' not in params:
             raise serializers.ValidationError(
                 _(u'Your password is required when updating your email '
                   u'address.'))
+        password = params.get("password1")
```
Comment (pld) [inline onadata/libs/serializers/user_profile_serializer.py]: if you move this above line 151, you can have the second conjuct in that conditional be `password`

File: onadata/libs/serializers/user_profile_serializer.py [hunk 2]
```diff
@@ -167,7 +170,7 @@ def update(self, instance, validated_data):
         instance.user.save()
 
         # force django-digest to regenerate its stored partial digests
-        update_partial_digests(instance.user, params.get("password1"))
+        update_partial_digests(instance.user, password)
```
Comment (pld) [inline onadata/libs/serializers/user_profile_serializer.py]: There's an error here that needs a test, if you add a test that passes no `email` and no `password` in params this should execute with `password` set to `nil` and set the partial digests to a bad state.

I'd suggest writing that test and then getting it passing by adding if `password` here

Comment (denniswambua) [inline onadata/libs/serializers/user_profile_serializer.py]: This has been addressed

File: onadata/libs/serializers/user_profile_serializer.py (hunk 1 above)
Comment (denniswambua) [inline onadata/libs/serializers/user_profile_serializer.py]: addressed

**Comment by skambo** [comment]
@denniswambua Theres validation, but the error msg is displayed as below when incorrect..

screen shot 2016-10-06 at 12 41 52 pm

Error when password is left blank

screen shot 2016-10-06 at 12 44 29 pm

**Comment by pld** [comment]
That part is a zebra problem in handling the API response, this branch is
good to merge.
